tests/beaker_tests/cisco_acl/test_ace.rb

```diff
@@ -0,0 +1,174 @@
+#
+# These keys are local use only and not used by test_harness_common:
+#
+# tests[id][:manifest_props] - This is essentially a master list of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the list
+# tests[id][:resource_props] - This is essentially a master hash of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#   Can be used with :af for mixed title/af testing. If mixing, :af values will
+#   be merged with title values and override any duplicates. If omitted,
+#   :title_pattern will be set to 'id'.
+# tests[id][:af] - (Optional) defines the address-family values.
+#   Must use :title_pattern if :af is not specified. Useful for testing mixed
+#   title/af manifests
+#
+tests['create_ipv4_seq_10'] = {
+  title_pattern:  'ipv4 foo-1 10',
+  manifest_props: "
+    'action'        => 'permit',
+    'proto'         => 'tcp',
+    'src_addr'      => '1.2.3.4 2.3.4.5',
+    'src_port'      => 'eq 40',
+    'dst_addr'      => '9.9.0.4/32',
+    'dst_port'      => 'range 32 56',
+  ",
+}
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = PUPPET_BINPATH + 'resource cisco_ace'
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_ace(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    manifest = tests[id][:manifest_props_absent]
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug("build_manifest_ace :: title_pattern:\n" +
+               tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'agent-lab3-nx' {
+    cisco_ace { '#{tests[id][:title_pattern]}':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_ace(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_ace(tests, id)
+
+  test_harness_common(tests, id)
+  # add case
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------------------------------------------------------------
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+  # node_feature_cleanup(agent, 'acl')
```

Comment:
1. There is no feature acl to cleanup so this should be removed.
2. You should have some kind of testbed cleanup that makes sure the acl is removed.
3. If I ran this test on my own testbed it would fail because there's nothing in this test that creates the ACL that this ACE needs to be added to; ie. when your `build_manifest_ace` is called it needs to also make a `cisco_acl` entry as part of the manifest.